### ISSUE TITLE
how to hook require in this format? eg.require('react-dom/server')

### ISSUE THREAD

**Issue by yuxizhe** [elastic/apm-agent-nodejs#913]
I want to add a module to support React server side render,  shimmer the react render time. 

![image](https://user-images.githubusercontent.com/14990734/54004445-84c35280-4190-11e9-9ef1-322b7e1f8bf9.png)

but react-dom-server is required in different format `"require('react-dom/server')"` .  I cannot get the right package, only get `react-dom`.  
[https://reactjs.org/docs/react-dom-server.html](react-server-side)

but `renderToString` is a function of `react-dom/server` other than `react-dom`

so I get error

```
no original function renderToString to wrap
```
I console.log other info in require-in-the-middle, maybe could help
```
path:   /Users/xxxxxxx/node_modules/react-dom/cjs/react-dom-server.node.development.js
stat:   { name: 'react-dom',
  basedir: '/Users/yu/dj-web/node_modules/react-dom',
  path: 'cjs/react-dom-server.node.development.js' }
res:  /Users/xxxx/node_modules/react-dom/index.js
shimming react-dom@16.8.2 module
```

my module file show blow
```js
'use strict'

var shimmer = require('../shimmer')

module.exports = function (reactDomServer, agent, version, enabled) {
  if (!enabled) return reactDomServer
  shimmer.wrap(reactDomServer, 'renderToString', wrapCompile)

  return reactdom

  function wrapCompile (original) {
    return function wrappedCompile (input) {
      var span = agent.startSpan('react SSR render', 'react')
      var id = span && span.transaction.id

      agent.logger.debug('intercepted call to react renderToString %o', {
        id: id,
        input: input
      })

      var ret = original.apply(this, arguments)
      if (span) span.end()

      return wrapTemplate(ret)
    }
  }
}
```

**Comment by yuxizhe** [comment]
@watson Amazing!  I never thought it would be solved so quickly and completely.

**Comment by watson** [comment]
The ability to patch sum-modules have landed in master now and will be part of the next release

**Comment by watson** [comment]
The feature was just released in v2.7.0 😊

Check out the new documentation for adding custom patches here: https://www.elastic.co/guide/en/apm/agent/nodejs/current/agent-api.html#apm-add-patch
